dissect/target/tools/query.py

```diff
@@ -110,67 +118,127 @@ def main():
                 "The --rewrite-cache option will be ignored as --no-cache or --only-read-cache are specified",
             )
 
+    # -l/-f follow the same code path, a list = a list of functions + a display command
+    # this makes sure the behavior will always be in sync
+    if args.list is not False:
+        args.function = args.list
+
+    # If no functions (or list) then we cannot continue at all
+    if not args.function:
+        parser.error("argument -f/--function is required")
+
+    # No targets is allowed, but only for lists/help
+    if not args.targets and args.list is False and "-h" not in rest and "--help" not in rest:
+        parser.error("too few arguments")
+
+    # In that case the user wants to see all available plugins and loaders
+    # and we turn off compatibility checks, because they will always fail
+    # we don't raise an exception though, because the user might want to 'amend'
+    # the command later...
+    if not args.targets:
+        targets = [Target()]
+        args.compatible = False
+    else:
+        targets = Target.open_all(args.targets, args.children)
+
+    # To check whether the output is uniform and which plugins we need to load
+    # either for the list or for execution, we collect all the work to be done here
+    tasklist = []
+    for target in targets:
+        if args.child:
+            try:
+                target = target.open_child(args.child)
+            except Exception:
+                continue
+        functions = target.find_plugin_functions(args.function, args.compatible)
+        tasklist.append((target, functions))
+
+    # Show the list of available plugins for the given optional target and optional
+    # search pattern, only display plugins that can be applied to ALL targets
     if args.list:
+        collected_plugins = None
+        for index, task in enumerate(tasklist):
+            collected = {}
+            target, funcs = task
+            for func in funcs:
+                collected[func.name] = func.plugin_desc
```

Comment:
Why this extra step with `collected`? Dict values are unique anyway, so just
```suggestion
                collected_plugins[func.name] = func.plugin_desc
```
should work.

Reply:
Ah, I guess because of the constraint that a plugin should be executable on ALL of targets and their children. But I think that is not what we want, it should be ANY target and their children.